README.md

```diff
@@ -1 +1,144 @@
+3. Authenticate the [flows.network](https://flows.network/) to access the `rfq-marketplace` repo you just forked. 
+
+4. Click on the Advanced text and you will see more settings including branch and environment variables. The function code is stored in `flows` folder, you need to change Directory to `/flows`.
+In this example, we have three variables to fill in, `CONTRACT_ADDRESS` is rfq-marketplace contract [address]( #deploy-the-contract).
+The default network is cybermiles. If you want to change the network, you can set `RPC_NODE_URL` and `CHAIN_ID` variable.
+
+<img width="899" alt="image" src="https://i.imgur.com/xSAxwLF.png">
+
+5. Click the Deploy button to deploy your function.
+
+### Configure SaaS integrations
+
+After that, the flows.network will direct you to configure the SaaS integration required by your flow. Here we can see: there is no SaaS needs to be connected since it's a lambda service. Just click the Check button to see your flow details.
+
+<img width="964" alt="image" src="https://user-images.githubusercontent.com/45785633/226959151-0e8a159a-02b3-4130-b7b5-8831b65c8d75.png">
+
+## Try this demo
+### Create an exchange request
+
+You can use the `submit-request` function to create a new exchange request. You need to provide three query parameters.<br>
+`tokenA`: The token address you want to exchange out. <br>
+`tokenB`: The token address you want to exchange in. <br>
+`amount`: The amount of tokenA you want to exchange.<br>
+In the demo, we use curl to send the post request, you can copy and paste the endpoint URL to your shell and add `/submit-request?tokenA=0x30D30c71d8618Ce42783eDd2C7Ae6f15eeD69Fec&tokenB=0x948Fa9010EFBEed5f4943893a383B7e2210bA145&amount=100`. <br>
+(You need to approve enough tokens to rfq contract before creating an exchange request)
+
+``` shell
+curl -X POST "https://code.flows.network/webhook/ekbbxC47MjjtIaP8RmO8/submit-request?tokenA=0x30D30c71d8618Ce42783eDd2C7Ae6f15eeD69Fec&tokenB=0x948Fa9010EFBEed5f4943893a383B7e2210bA145&amount=100" \
+-d '{"PRIVATE_KEY": "Exchange request owner private key"}'
+```
+Then you can see the transaction result.<br>
+
+### List requests
+
+After creating an exchange request, use the `list-requests` function to query all the exchange requests.<br>
+Copy and paste the endpoint URL to your browser and add `/list-requests`.
+Then you can see all the exchange requests in the rfq-marketplace.
+
+<img width="964" alt="image" src="https://i.imgur.com/4wJWy9n.png">
+
+### Response exchange
+
+If you want to exchange tokens with others, use `submit-response` to respond to other's exchange requests.<br>
+You can copy and paste the endpoint URL to your shell and add `/submit-response?request-id=0&amount=10`.
+(You need to approve enough tokens to rfq contract before the response exchange)
+
+``` shell
+curl -X POST "https://code.flows.network/webhook/ekbbxC47MjjtIaP8RmO8/submit-response?request-id=0&amount=10" \
```

Comment:
Does this mean that the buyer offered 10 `token_in` in exchange of 100 `token_out` from the requester?

Reply:
I think that we need to add a feature here as well: the response should have an expiration date. It could be in UTC seconds. When the expiration date is over, the response will not show up in the query list and cannot be executed.

Reply:
Yes, I have add more description and add the feature. Please check, thank you.